### PR TITLE
Add association between id fields

### PR DESCRIPTION
**Pull request by LongfeiH**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] [local build the book](https://github.com/ocademy-ai/machine-learning/blob/main/CONTRIBUTING.md#build-the-book)
- [ ] review the artifacts from local build
- [ ] [local build the slides](https://github.com/ocademy-ai/machine-learning/blob/main/CONTRIBUTING.md#build-the-slides-optional) if appliable
- [x] Unit test if appliable

# Checklist:

- [x] My code follows the [style guidelines](https://github.com/ocademy-ai/machine-learning/blob/main/STYLE_GUIDE.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have applied tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: awesome/database/migrations/20230820065806_020_insert_directus_fields_UserCourses_courseId.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/**
```
Comment (zhangqi444) [inline awesome/database/migrations/20230820065806_020_insert_directus_fields_UserCourses_courseId.js:1]: All such code is duplicated. Could you refer to this? https://github.com/ocademy-ai/machine-learning/blob/main/awesome/database/migrations/20230815040913_012_insert_directus_fields_constrain_for_ids.js

You can create a helper function in utils.js as `addDirectusFKInterface(collection, field)`

File: awesome/database/migrations/20230820042135_014_turn_off_readonly_and_hidden_of_id_field.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = function(knex) {
+  const idsToUpdate = [46, 47, 48, 49, 50, 51, 52, 53, 54, 55];
```
Comment (zhangqi444) [inline awesome/database/migrations/20230820042135_014_turn_off_readonly_and_hidden_of_id_field.js:6]: You CANNOT use id to query here. It is not reliable. Please use `field` and `collection` to filter out the rows to update. Please refer to https://github.com/ocademy-ai/machine-learning/blob/main/awesome/database/migrations/20230815224628_013_insert_directus_fields_constrain_for_createdAt_and_updatedAt.js

File: awesome/database/migrations/20230820065806_020_insert_directus_fields_UserCourses_courseId.js (hunk 1 above)
Comment (LongfeiH) [reply awesome/database/migrations/20230820065806_020_insert_directus_fields_UserCourses_courseId.js]: done

File: awesome/database/migrations/20230820042135_014_turn_off_readonly_and_hidden_of_id_field.js (hunk 2 above)
Comment (LongfeiH) [reply awesome/database/migrations/20230820042135_014_turn_off_readonly_and_hidden_of_id_field.js]: done

File: awesome/database/utils.js [hunk 3]
```diff
@@ -116,7 +116,38 @@ async function insertColumn(knex, tableName, columnName, columnType) {
     knex.destroy();
   }
 }
-
+// insert directus field
+async function addDirectusFKInterface(knex, tableName, collection, field) {
```
Comment (zhangqi444) [inline awesome/database/utils.js:120]: Add empty line before and after each function.

**Review by zhangqi444 — APPROVED** [review]

**Pull request merged**